src/main/java/jnr/ffi/util/ref/internal/Finalizer.java

```diff
@@ -244,6 +244,12 @@ private Method getFinalizeReferentMethod() {
 
 
     public static Field getInheritableThreadLocalsField() {
+        String javaVersion = System.getProperty("java.specification.version");
+        if (Integer.valueOf(javaVersion.split(".")[0]) >= 9) {
```

Comment:
It's a shame that sniffing strings is still the best thing we can do for a Java version check. This doesn't break for the new version scheme `18.3` does it?

Reply:
I think it should be fine as we check the first number before the `.` (or just the only number if there is no dot).
But I'm open to alternatives if there are better ways.
There is [Runtime.version()](http://download.java.net/java/jdk9/docs/api/java/lang/Runtime.html#version--) but that's only available in Java 9.

Reply:
I'm planning to make a similar change upstream, but I'll also be invoking the new Thread constructor that has a boolean inheritThreadLocals parameter so that we still get the functionality.

Reply:
:+1: In jnr-ffi, the logic is already to use that constructor:
https://github.com/eregon/jnr-ffi/blob/a8260ce70c8d123a31a5285147d34cee1224a391/src/main/java/jnr/ffi/util/ref/internal/Finalizer.java#L132-L141

Reply:
I have made the change now, and what I did was to look up the Constructor first, and only look up the Field if the Constructor lookup failed. That avoids having to put in a version check.

Reply:
That's a nice way to do it without needing the version check in https://github.com/google/guava/commit/2927518ac86e8ed1d09038fb2b03c86c894ea5ca, I'll do something similar.

Reply:
Thanks for the idea, I applied the same idea in this PR as well.